pkg/userdata/helper/kubelet.go

```diff
@@ -317,6 +317,18 @@ func kubeletConfiguration(log *zap.SugaredLogger, clusterDomain string, clusterD
 		}
 	}
 
+	if maxParallelImagePulls, ok := kubeletConfigs[common.MaxParallelImagePullsKubeletConfig]; ok {
+		val, err := strconv.ParseInt(maxParallelImagePulls, 10, 32)
+		if err != nil || val < 0 {
```

Comment:
Now it is valid to set maxParallelImagePulls to 0 which either throws an error or blocks all images being pulled I would assume.
I think it is better to make sure that the value is at least 1